lib/libefi/rdwr_efi.c

```diff
@@ -1167,16 +1212,13 @@ efi_use_whole_disk(int fd)
 		return (VT_ENOSPC);
 	}
 
-	/*
-	 * Find the last physically non-zero partition before that.
-	 * This is the data partition.
-	 */
-	for (i = 0; i < resv_index; i ++) {
-		if (data_start < efi_label->efi_parts[i].p_start) {
-			data_start = efi_label->efi_parts[i].p_start;
-			data_index = i;
-		}
-	}
+	VERIFY3U(data_start + data_size, ==, resv_start);
+	VERIFY3U(limit, >=, resv_start);
```

Comment:
nit: Technically I believe we should be using `verify()` here for portability since this is built solely as a library.  That said, returning `VT_EINVAL` as an error would probably be better since we are working with values read from disk and the vtoc could be damaged/incorrect.  It would be better to handle the error rather than crash.

Reply:
I can try to make this change if you'd like it in this PR..